### DISCUSSION TITLE
Turning off/on crazyflie's motors

### DISCUSSION THREAD

**Discussion by xiaoshan-lin** [bitcraze/discussions#1182] [Q&A]
**High-level Problems:** We want to have our crazyflies repeatedly execute the following actions: take off -> land -> turn off motors -> turn on motors -> take off -> ... . 

Using the land() function will prevent the crazyflie from taking off again. So we implemented

-    a "soft-landing" function that includes: 

       -  goTo the landing position
       -  turn off the motors

-    and a "soft_takeoff" function that:

       - turn on the motors
       - goTo the takeoff position

**What is currently working:** As discussed in https://github.com/IMRCLab/crazyswarm2/discussions/418, we are able to turn off and turn on the motors by setting the parameters "stabilizer.stop" with **firmware 22.05**
```
    allcfs.setParam("stabilizer.stop", 1) # turn off motors
    allcfs.setParam("kalman.resetEstimation", 1)
    timeHelper.sleep(2.0)

    allcfs.setParam("stabilizer.stop", 0) # turn on motors
    timeHelper.sleep(2.0)

    for idx, cf in enumerate(allcfs.crazyflies):
        cf.goTo(takeoff_pos[idx], 0, duration=duration_to_take_off)  # take off
    timeHelper.sleep(3.0)
```

**What is not working:** 

-  With **firmware 23.11**, we are unable to turn on motors by allcfs.setParam("stabilizer.stop", 0) and turn off motors by allcfs.setParam("stabilizer.stop", 1). 
- In previous [discussion](https://github.com/IMRCLab/crazyswarm2/discussions/418) with maintainers of crazyswarm2, they suggested setting the parameter "motorPowerSet". We are able to turn on the motors by
   ```
    cf.setParam("motorPowerSet.m1", 40000)
    cf.setParam("motorPowerSet.m2", 40000)
    cf.setParam("motorPowerSet.m3", 40000)
    cf.setParam("motorPowerSet.m4", 40000)
    cf.setParam("motorPowerSet.enable", 1)
   ```
   But the crazyflie doesn't respond to high-level commands such as goTo().
- 

**Questions: (for firmware 23.11)**
1. What is the recommended way of temporarily turning off the motors?
2. How to turn on the motors and make sure the crazyflie can respond to high-level commands, such as goTo()?

**Comment by tobbeanton** [comment]
I think we introduced the supervisor and arming in the later firmware. So by [arming and disarming](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/supervisor/arming/) the crazyflies you can achieve what you want but you also have to build the firmware with [auto-arming](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/supervisor/arming/#auto-arming) disabled.

**Reply by xiaoshan-lin** [reply]
Thanks for your quick response. 

To disable the auto-arming when compiling, should I add CONFIG_MOTORS_REQUIRE_ARMING=n to the cf2_defconfig file? Or should I customize it using `make menuconfig` ?

**Reply by tobbeanton** [reply]
I would say altering the cf2_defconfig would be the wrong way but you could create a new config file, that might actually be the best way, so you can easily come back to that config. Else `make menuconfig` works too.

**Reply by tobbeanton** [reply]
And the setting should be yes `CONFIG_MOTORS_REQUIRE_ARMING=y`

**Reply by wladimirpetrov** [reply]
Hello! 

I've been trying to follow similar approach. I have a "soft" landing and take off, where in landing I turn the motors off and reset kalman estimation, in take off I turn on the 
 

I did: 
`nano configs/cf2_defconfig` 
and inside i added: 
`CONFIG_MOTORS_REQUIRE_ARMING=y`.

But now the crazyflie doesn't want to take off when I do

```
cf.takeoff(targetHeight=0.3, duration=TAKEOFF_DURATION)
timeHelper.sleep(TAKEOFF_DURATION + HOVER_DURATION)
```
My understanding of this is that, as far as I turned off the auto arming, then I should arm and disarm crazyflie by myself.
I followed the links you provided above about [arming and disarming](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/supervisor/arming/#auto-arming), but I don't think I understood what line i should put in my python script to be able to arm/disarm crazyflie. 

Could you please help me out with this issue?
Thank you for your time and assistance!

**Reply by ArisMorgens** [reply]
Hello Vladimir!

Arming/disarming the Crazyflie can be done in a python script through the  `cf.platform.send_arming_request()` command. You can check [this](https://github.com/bitcraze/crazyflie-lib-python/blob/eb22acc8bb8ccf654426203e952d41343f1f76de/examples/step-by-step/sbs_motion_commander.py#L128-L130) example as a reference. 
Does this answer your question?

**Comment by wladimirpetrov** [comment]
Thank you, sorry for late response, I'll definitely try this

**Reply by gemenerik** [reply]
Please let us know how it goes!
